### PR TITLE
Example

### PR DESCRIPTION
**Pull request by yalh76**
## Solution
- *Apply example_ynh*

### CODE REVIEW

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
#Own CI result for Revision: ebcd563647c12996949f21b3ecf4f75e4178de86 

image

File: scripts/install [hunk 1]
```diff
@@ -162,27 +161,25 @@ done
 #=================================================
 # SECURE FILES AND DIRECTORIES
 #=================================================
+ynh_script_progression --message="Securing files and directories..."
 
 # Set permissions to app files
-chown -R root: $final_path
-chown pepettes:root $final_path
-chown pepettes:root $final_path/settings.py
-chmod o=--- $final_path/settings.py
```
Comment (zamentur) [inline scripts/install:170]: How we avoid to expose this files to others users (like potential hacked apps)

File: scripts/restore [hunk 2]
```diff
@@ -63,14 +64,15 @@ ynh_restore_file --origin_path="$final_path"
...
+ynh_system_user_create --username=$app --home_dir="$final_path"
 
 #=================================================
 # RESTORE USER RIGHTS
 #=================================================
+ynh_script_progression --message="Restoring user rights..."
 
 # Restore permissions on app files
-chown -R root: $final_path
+chown -R $app:root $final_path
```
Comment (zamentur) [inline scripts/restore:75]: chmod o=--- missing too

File: scripts/upgrade [hunk 3]
```diff
@@ -79,56 +93,84 @@ ynh_add_nginx_config
...
 #=================================================
 # GENERIC FINALIZATION
 #=================================================
 # SECURE FILES AND DIRECTORIES
 #=================================================
+ynh_script_progression --message="Securing files and directories..."
 
 # Set permissions on app files
-chown -R root: $final_path
+chown -R $app:root $final_path
```
Comment (zamentur) [inline scripts/upgrade:159]: chmod o=--- missing too

File: scripts/upgrade [hunk 4]
```diff
@@ -79,56 +93,84 @@ ynh_add_nginx_config
 ynh_script_progression --message="Upgrading dependencies..." --weight=1
 
 ynh_install_app_dependencies $pkg_dependencies
+ynh_install_python --python_version=$python_version
```
Comment (zamentur) [inline scripts/upgrade:96]: It's enough to upgrade python ?

Is there a risk that a new python version with hotfix appears and this package was not updated quickly (compared to debian package) ?

**Review by zamentur — CHANGES_REQUESTED** [review]

File: scripts/install (hunk 1 above)
Comment (yalh76) [reply scripts/install:170]: Did a `chmod -R 750` or `chmod -R 770` would be enough ?
- `chmod -R 750` : https://chmodcommand.com/chmod-750/
- `chmod -R 770` : https://chmodcommand.com/chmod-770/
In that case only the $app has access and also root group

File: scripts/upgrade (hunk 4 above)
Comment (yalh76) [reply scripts/upgrade:96]: > 
> 
> It's enough to upgrade python ?

Yes it's enough `ynh_install_python`: 

1. install or update pyenv
2. install requested `$python_version`
3. recreate an `$app` virtualvenv based on `$python_version`
4. cleanup no more needed python version

> 
> Is there a risk that a new python version with hotfix appears and this package was not updated quickly (compared to debian package) ?

For now, Debian 10 is provided with:
- Python 2.7.16 (March 4, 2019) : latest official is 2.7.17 (Oct. 19, 2019)
- Python 3.7.3 ( March 25, 2019) : latest is 3.7.10 (Feb. 15, 2021)
- There is also 3.8.8 (Feb. 19, 2021) and 3.9.2 (Feb. 19, 2021)

So, I'm pretty sure that yunohost python will be up to date ^^ not 2 years old....

**Comment by yalh76** [comment]
# Own CI result for Revision: c35da86ffac5d0cdf8f9fd61de54868055c005b4 

image

I will look what's happen

File: scripts/upgrade (hunk 4 above)
Comment (alexAubin) [reply scripts/upgrade:96]: To me it ain't clear what's the benefit of installing a specific version of Python here ... I mean Buster already ships Python 
3.7.3 (= `$python_version` ?). Python is supposed to be forward-compatible (with the exception of Python 2 vs. 3 of course) sooooooo ...?

I mean from what I know of pepettes it should be super-lightweight but compiling Python locally adds a huge cost ? :s

Comment (alexAubin) [reply scripts/upgrade:96]: (Source: i'm typing this while waiting for the job on ci-apps-dev to finish while it compiles python for the 10th time :/ ...)

Comment (yalh76) [reply scripts/upgrade:96]: I fully agree that for pepettes, having `ynh_install_python` is overkilled... It's more a proof of concept.. As pepettes is small, it's quicker to test on it.

**Comment by yalh76** [comment]
# Own CI result for Revision: 1a0f985e5175040d63b38862a9e2889742e4aaa6 

image

File: scripts/install [hunk 5]
```diff
@@ -162,27 +161,26 @@ done
...
+yunohost service add $app --description="A simple donation form" --log_type="systemd"
 
 #=================================================
 # START SYSTEMD SERVICE
 #=================================================
 ynh_script_progression --message="Starting a systemd service..." --weight=1
 
 # Start a systemd service
-ynh_systemd_action --service_name=$app --action="start" --log_path="/var/log/$app/$app.log"
+ynh_systemd_action --service_name=$app --action="start" --log_path="systemd" --line_match="Started"
```
Comment (alexAubin) [inline scripts/install:183]: Not sure to understand why that doesn't work, but on ci-apps-dev : 

```
333744 INFO [###############+....] > Starting a systemd service...
635556 WARNING Please wait, the service pepettes is starting..........................................................................................................................................................................................................................................................................................................
635558 WARNING The service pepettes didn't fully executed the action start before the timeout.
635558 WARNING Please find here an extract of the end of the log of the service pepettes:
635564 WARNING Mar 10 23:35:53 systemd[1]: Started pepettes Daemon.
```

i.e. it doesn't find the "Started" despite the fact that it's in the systemd journal :/

Comment (yalh76) [reply scripts/install:183]: ... It's something I've already seen and that we discuss at:
- https://github.com/YunoHost/yunohost/pull/865
- https://github.com/YunoHost/yunohost/pull/959
- https://github.com/YunoHost/issues/issues/1573

Related to ynh_systemd_action not catching the first lines of log when the service start too quickly ...

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
!testme

**Comment by yalh76** [comment]
# Own CI result for Revision: 1c5bf636b3103d934e5767b052861134f6364dd6 
image

**Comment by yalh76** [comment]
!testme

File: scripts/upgrade (hunk 4 above)
Comment (yalh76) [reply scripts/upgrade]: ynh_install_python removed

**Review by alexAubin — APPROVED** [review]
Super quick review, LGTM :+1:

**Pull request merged**